service/http.ts

```diff
@@ -45,10 +66,18 @@ export async function post<T, U>(
   return await http<U>(path, init);
 }
 
-export async function put<T, U>(
+export async function authenticatedPost<T, U>(
```

Comment:
Instead of naming this `authenticatedPost`, I wonder if it would be nicer to stick to a single `post` (same comment apply to other HTTP method helper functions), with a signature looks something like the following:

```typescript
// I find the single letter generic confusing and prefers more explicit ones
// prefixed with T, inspired from react-query if you ever look at their type definitions
function post<TBody, TResponse>(
  path: string,
  config?: { requestBody: TBody; requiresAuth: boolean }
) {
  // Use http or authenticatedHttp and set the relevant config object for `fetch` depending
  // on if `requiresAuth` is true or not
}